src/main/java/seedu/clinic/logic/commands/AddCommand.java

```diff
@@ -66,27 +82,36 @@ public AddCommand(Warehouse warehouse) {
         requireNonNull(warehouse);
         this.supplierToAdd = null;
         this.warehouseToAdd = warehouse;
-
+        logger.log(Level.INFO, "Received information to add warehouse");
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+        CommandResult commandResult;
 
         if (supplierToAdd != null) {
             if (model.hasSupplier(supplierToAdd)) {
                 throw new CommandException(MESSAGE_DUPLICATE_SUPPLIER);
             }
             model.addSupplier(supplierToAdd);
-            return new CommandResult(String.format(MESSAGE_SUPPLIER_SUCCESS, supplierToAdd));
-        } else if (warehouseToAdd != null) {
+            logger.log(Level.INFO, "Supplier with given information has been added and supplier list is"
+                    + " updated on UI.");
+
+            commandResult = new CommandResult(String.format(MESSAGE_SUPPLIER_SUCCESS, supplierToAdd));
+        } else {
+            assert this.warehouseToAdd != null : "warehouseToAdd specified"
```

Comment:
I wrote it the way as you mentioned initially, then intellij prompted me that it will always return true and thus suggested to change to `!= null instead`. I'm not sure which is better coding practice but I could change it back and ignore the warning if we standarize it across out code